tests/test_aea.py

```diff
@@ -484,8 +484,9 @@ def test_add_behaviour_dynamically():
     resources = Resources()
     identity = Identity(agent_name, address=wallet.addresses[DEFAULT_LEDGER])
     connection = _make_local_connection(identity.address, LocalNode())
-    agent = AEA(identity, wallet, resources, default_connection=connection.public_id,)
     resources.add_connection(connection)
```

Comment:
So we now enforce that `resources` is complete at time of construction of AEA.

Reply:
@DavidMinarsch  at least connections

Reply:
I will add a proper check  on agent init

Reply:
done